tensorflow_addons/metrics/r_square.py

```diff
@@ -130,6 +130,9 @@ def result(self) -> tf.Tensor:
         mean = self.sum / self.count
         total = self.squared_sum - self.sum * mean
         raw_scores = 1 - (self.res / total)
+        raw_scores = tf.where(
+            tf.math.is_inf(raw_scores), tf.zeros_like(raw_scores), raw_scores
```

Comment:
`tf.where`  supports broadcast. We can use `0` instead of  `tf.zeros_like(raw_scores)` to avoid materialize larger constant.

Reply:
Yes I don't know if probably `tf.where(tf.math.is_inf(raw_scores),0, raw_scores)` could be ok.

Reply:
Clever!

Reply:
Uhm, this change appears to break some tests...

Reply:
You need `0.`